engine.go

```diff
@@ -30,11 +31,11 @@ type Config struct {
 type Engine struct {
 	sync.Mutex
 	latest   uint64
-	provider types.Provider
+	provider jsonrpc.Client
 	ticker   *time.Ticker
 }
 
-func NewEngine(ctx context.Context, provider types.Provider, config Config) (*Engine, error) {
+func NewEngine(ctx context.Context, provider jsonrpc.Client, config Config) (*Engine, error) {
```

Comment:
lets update `types.Provider` to match `jsonrpc.Client` and then we can modify the gateway provider to match that too